File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/inline/comment/direct/Java/Python/s185737640_commented.py
def is_even_string(s):
    half = len(s) // 2
    return s[:half] == s[half:]

s = input().strip()

if len(s) % 2 == 1:
    s = s[:-1]
else:
    s = s[:-2]

while s:
    if is_even_string(s):
        break
    s = s[:-2]

print(len(s))  # End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/inline/comment/direct/Java/C++/s185737640_commented.cpp
#include <iostream>
#include <string>
using namespace std;

bool isEvenString(const string& s) {
    size_t mid = s.length() / 2;
    string a = s.substr(0, mid);
    string b = s.substr(mid);
    return a == b;
}

int main() {
    string S;
    getline(cin, S);

    if (S.length() % 2 == 1) {
        S = S.substr(0, S.length() - 1);
    } else {
        S = S.substr(0, S.length() - 2);
    }

    while (true) {
        if (S.empty()) break;
        if (isEvenString(S)) break;
        S = S.substr(0, S.length() - 2);
    }

    cout << S.length() << endl;
    return 0;
}
// End of Code
